src/ml/surrogate.hpp

```diff
@@ -104,6 +106,30 @@ class SurrogateModel
     }
   }
 
+PERFFASPECT()
+  inline void tensorToHostArray(at::Tensor tensor,
+                            long numRows,
+                            long numCols,
+                            TypeInValue** array)
+  {
+    // Transpose to get continuous memory and
+    // perform single memcpy.
+    tensor = tensor.transpose(1, 0);
+    if (_is_cpu) {
+      for (long j = 0; j < numCols; j++) {
+        auto tmp = tensor[j].contiguous();
+        TypeInValue* ptr = tmp.data_ptr<TypeInValue>();
+        HtoHMemcpy(array[j], ptr, sizeof(TypeInValue) * numRows);
+      }
+    } else {
+      for (long j = 0; j < numCols; j++) {
+        auto tmp = tensor[j].contiguous();
+        TypeInValue* ptr = tmp.data_ptr<TypeInValue>();
+        DtoHMemcpy(array[j], ptr, sizeof(TypeInValue) * numRows);
+      }
+    }
+  }
```

Comment:
What is the difference with `tensorToArray`?

Reply:
`tensorToHostArray` always copies data to the host, `tensorToArray` copies data to the originating device. More specifically, `tensorToHostArra` calls `DtoHMemcpy` to copy data from the device to the host for UQ predicate calculation